### DISCUSSION TITLE
SQL left join to code

### DISCUSSION THREAD

**Discussion by Tazi0** [supabase/supabase#27101] [Questions]
```sql
SELECT * FROM selections
LEFT JOIN results
  ON results."jointId" = selections."jointId" AND selections."driverId" = results."driverId"
```
So I have this project where I have 2 tables, selections and results. My goal is that I can read all the positions the player guessed correctly based upon their selection. I've built the above SQL query, but I can't seem to translate that into executable JS code. Can anyone assist me. I want my JSON to look like this at the end:
```
{
    name: "player 1",
    selections: [{
            driver: {...},
            position: 5
    }, ...]
}
```
My schema:
![image](https://github.com/supabase/supabase/assets/45716542/66b53526-a326-44f9-8aad-21051eb3bd9d)

Notes by me: So the 2 tables have a connecting table `seasons_to_races` which is a many-to-many relation for seasons & races. I used this table because every player selection & race result is only for a specific race in a specific reason. I could make another connection from the selection table to the results table to make it easier with selecting, but that would defeat the purpose of having the results table since it's not used otherwise, and I want to build upon this system.

I currently have a workaround which just calls supabase 2 times, but that isn't optimal for loading and the repeating requests I would have to make.

**Comment by Tazi0** [comment]
```
{
  code: 'PGRST200',
  details: "Searched for a foreign key relationship between 'selections' and 'results' in the schema 'public', but no matches were found.",
  hint: null,
  message: "Could not find a relationship between 'selections' and 'results' in the schema cache"
}
```
I switched the primary key of `seasons_to_races` to the unique ID, but that didn't seem to help :/

**Comment by GaryAustin1** [comment]
The API requires foreign key links for all joins.
If you are using a join table you need to make sure both foreign keys are marked as part of the primary "key".
There is also this new tool you might try.  https://supabase.com/docs/guides/api/sql-to-rest

**Reply by Tazi0** [reply]
Hi Gary! they are part of the primary key. Actually, the entire primary key should be the same 🤔 

<img width="651" alt="Screenshot 2024-06-07 at 21 54 45" src="https://github.com/supabase/supabase/assets/45716542/eff61844-c3f0-46bd-8f76-923154149a28">

**Reply by GaryAustin1** [reply]
Seems like the issue is you want both driverId and jointId as the join condition.  PostgREST the API is not going to be able to handle that directly.  It wants to use AN fk (even thru other tables) to find a relationship.  
The tool I linked will let you play and flag syntax that works or doesn't (very new tool though).

I'm not looking at your structure for other possibilities to do the relation though.

You can also just use a view.

**Reply by Tazi0** [reply]
Right, I've tried the tool for a little and understand what the problem is. Sadly a different key configuration wouldn't be possible since there are multiple drivers in a particular race, therefor I cannot only set the `jointId` as primary key.

I'll try looking into a view!

**Comment by phpusrpro** [comment]
Check this https://preprocessor.space/questions/tips-for-emulating-the-sql-left-join-function-with-php-arrays
